pkg/vere/io/http.c

```diff
@@ -443,6 +450,24 @@ _http_req_unlink(u3_hreq* req_u)
   }
 }
 
+/* _http_spin_link(): store spin stack request in state
+*/
+static void
+_http_spin_link(u3_hcon* hon_u, u3_hreq* req_u)
+{
+  u3_hfig* fig_u = &hon_u->htp_u->htd_u->fig_u;
+  req_u->hon_u = hon_u;
+  req_u->seq_l = hon_u->seq_l++;
+  req_u->nex_u = fig_u->siq_u;
+
+  if ( 0 != req_u->nex_u ) {
+    req_u->nex_u->pre_u = req_u;
+  } else {
+    uv_timer_start(fig_u->sin_u, _http_spin_timer_cb, SPIN_TIMER, 0);
+  }
+  fig_u->siq_u = req_u;
```

Comment:
Here we start the timer if the request list was empty. `_http_spin_timer_cb` doesn't set a new timer if the request list is empty. But `_http_spin_unlink` doesn't _unset_ the timer. So `_http_spin_link` could get called while the request list is empty _and_ there's still an outstanding timer.

Does `uv_timer_start` protect against overwriting `sin_u`? If not, we might start a second timer (if a request comes in between the unlink and the og timer firing) that'll propagate itself alongside the og timer through the callback logic.

Reply:
`uv_timer_start` overwrites a pending timer for the handle if one is already pending, so it'll just be one timer no matter what.